packages/interface/src/features/applications/components/ApplicationsToApprove.tsx

```diff
@@ -0,0 +1,242 @@
+import Link from "next/link";
+import { useMemo, useEffect, useState } from "react";
+import { useFormContext } from "react-hook-form";
+import { FiAlertCircle } from "react-icons/fi";
+import { z } from "zod";
+
+import { EmptyState } from "~/components/EmptyState";
+import { Badge } from "~/components/ui/Badge";
+import { Button } from "~/components/ui/Button";
+import { Checkbox, Form } from "~/components/ui/Form";
+import { Heading } from "~/components/ui/Heading";
+import { Skeleton } from "~/components/ui/Skeleton";
+import { Spinner } from "~/components/ui/Spinner";
+import { useApplications } from "~/features/applications/hooks/useApplications";
+import { type Application } from "~/features/applications/types";
+import { ProjectAvatar } from "~/features/projects/components/ProjectAvatar";
+import { useIsAdmin } from "~/hooks/useIsAdmin";
+import { useIsCorrectNetwork } from "~/hooks/useIsCorrectNetwork";
+import { useMetadata } from "~/hooks/useMetadata";
+import { fetchApprovedApplications } from "~/utils/fetchAttestationsWithoutCache";
+import { formatDate } from "~/utils/time";
+
+import type { Attestation } from "~/utils/types";
+
+import { useApproveApplication } from "../hooks/useApproveApplication";
+import { useApprovedApplications } from "../hooks/useApprovedApplications";
+
+export interface IApplicationItemProps extends Attestation {
+  isApproved?: boolean;
+  isLoading?: boolean;
+}
+
+const SelectAllButton = ({ applications = [] }: { applications: Attestation[] | undefined }) => {
+  const form = useFormContext<TApplicationsToApprove>();
+  const selected = form.watch("selected");
+  const isAllSelected = selected.length > 0 && selected.length === applications.length;
+  return (
+    <Button
+      disabled={!applications.length}
+      type="button"
+      onClick={() => {
+        const selectAll = isAllSelected ? [] : applications.map(({ id }) => id);
+        form.setValue("selected", selectAll);
+      }}
+    >
+      {isAllSelected ? "Deselect all" : "Select all"}
+    </Button>
+  );
+};
+
+interface IApproveButtonProps {
+  isLoading?: boolean;
+}
+
+const ApproveButton = ({ isLoading = false }: IApproveButtonProps): JSX.Element => {
+  const isAdmin = useIsAdmin();
+  const { isCorrectNetwork, correctNetwork } = useIsCorrectNetwork();
+  const form = useFormContext<TApplicationsToApprove>();
+  const selectedCount = Object.values(form.watch("selected")).filter(Boolean).length;
+
+  const text = isAdmin ? `Approve ${selectedCount} applications` : "You must be an admin";
+
+  return (
+    <Button
+      suppressHydrationWarning
+      disabled={!selectedCount || !isAdmin || isLoading || !isCorrectNetwork}
+      size="auto"
+      type="submit"
+      variant="primary"
+    >
+      {!isCorrectNetwork ? `Connect to ${correctNetwork.name}` : text}
+    </Button>
+  );
+};
+
+const ApplicationHeader = ({ applications = [] }: { applications: Attestation[] | undefined }): JSX.Element => (
```

Comment:
Related to all cases

```suggestion
interface IApplicationBlockProps {
  applications?: Attestation[];
}

const ApplicationHeader = ({ applications = [] }: IApplicationHeaderProps): JSX.Element => (
```

Reply:
ok, let me change all of them